backend/src/test/java/gov/cdc/usds/simplereport/idp/authentication/LiveOktaAuthenticationTest.java

```diff
@@ -0,0 +1,517 @@
+            });
+    assertThat(exception).hasMessage("Error setting user's password");
+  }
+
+  @Test
+  void setRecoveryQuestionFails_withInvalidQuestion() {
+    Exception exception =
+        assertThrows(
+            OktaAuthenticationFailureException.class,
+            () -> {
+              _auth.setRecoveryQuestion(_userId, "Who was your third grade teacher?", "aa");
+            });
+    assertThat(exception).hasMessage("Error setting recovery questions");
+  }
+
+  @Test
+  void enrollSmsMfaFails_withInvalidPhoneNumber() {
+    Exception exception =
+        assertThrows(
+            OktaAuthenticationFailureException.class,
+            () -> {
+              _auth.enrollSmsMfa(_userId, "999");
+            });
+    assertThat(exception).hasMessage("Error setting SMS MFA");
+  }
+
+  @Test
+  void enrollVoiceCallMfaFails_withInvalidPhoneNumber() {
+    Exception exception =
+        assertThrows(
+            OktaAuthenticationFailureException.class,
+            () -> {
+              _auth.enrollVoiceCallMfa(_userId, "999");
+            });
+    assertThat(exception).hasMessage("Error setting voice call MFA");
+  }
+
+  @Test
+  void enrollEmailMfaFails_withInvalidEmail() {
+    Exception exception =
+        assertThrows(
+            OktaAuthenticationFailureException.class,
+            () -> {
+              _auth.enrollEmailMfa(_userId, "exampleemail");
+            });
+    assertThat(exception).hasMessage("Error setting email MFA");
+  }
+
+  @Test
+  void enrollAuthenticatorAppMfaFails_withInvalidAppType() {
+    Exception exception =
+        assertThrows(
+            OktaAuthenticationFailureException.class,
+            () -> {
+              _auth.enrollAuthenticatorAppMfa(_userId, "app");
+            });
+    assertThat(exception).hasMessage("App type not recognized.");
+  }
+
+  @Test
+  void verifyActivationPasscodeFails_withInvalidPasscode() {
+    FactorAndQrCode mfaResponse = _auth.enrollAuthenticatorAppMfa(_userId, "okta");
+    String factorId = mfaResponse.getFactorId();
+    Exception exception =
+        assertThrows(
+            OktaAuthenticationFailureException.class,
+            () -> {
+              _auth.verifyActivationPasscode(_userId, factorId, "1234");
+            });
+    assertThat(exception)
+        .hasMessage("Activation passcode could not be verifed; MFA activation failed.");
+  }
+
+  @Test
+  void resendActivationPasscodeFails_withUnenrolledFactor() {
+    Exception exception =
+        assertThrows(
+            OktaAuthenticationFailureException.class,
+            () -> {
+              _auth.resendActivationPasscode(_userId, "unenrolledFactorId");
+            });
+    assertThat(exception).hasMessage("An exception was thrown while fetching the user's factor.");
+  }
+
+  /**
+   * Helper method to create a user.
+   * https://developer.okta.com/docs/reference/api/users/#request-example
+   */
+  private JSONObject createUser() {
```

Comment:
Since this test class already has the Okta token and org url, I think it could just instantiate an Okta `Client` and use it for what `createUser`, `activateUser`, `deactivateUser`, etc are doing.  Achieves the same thing, but might make these helpers  unnecessary.

Reply:
Done - but a word of warning, this has made the diff look much worse than it actually is 😄  The test assertions haven't actually changed, I'm just asserting on Java objects instead of JSON now.